projects/interfaces/buffer.go

```diff
@@ -0,0 +1,42 @@
+package interfaces
+
+import (
+	"slices"
+	"errors"
+)
+
+//OurByteBuffer - Limitations
+// - the buffer grows and shrinks as data is added and read from it. The maximum data stored by the data is dependent on the machine in which the code is run. The buffer frees up all memory that has been successfully read from the buffer.
+//Both read and write operations modifies the buffer, hence it is unsafe to perform them concurrently. Retrieving bytes is also unsafe to run concurrently with either read or write operations.
+
+type OurByteBuffer struct {
+	buffer []byte
+}
+
+
+func (b *OurByteBuffer) Write(data []byte) (int, error) {
+	count := 0
+	for _, v := range data {
+		result := append(b.buffer, v)
+		b.buffer = result
+		count++
+	}
```

Comment:
append takes variable number of arguments and given that I know all I want to append before hand, I can do this instead 
`b.buffer = append(b.buffer, data...)`
With this, reallocation of memory if need be is done once instead of multiple times in the case of the for loop.